wagtail/search/backends/database/sqlite/sqlite.py

```diff
@@ -0,0 +1,607 @@
+                else:
+                    lexemes |= new_lexeme
+
+            return SearchQuery(lexemes, config=config)
+
+        elif isinstance(query, Phrase):
+            return SearchQuery(query.query_string)
+
+        elif isinstance(query, AndNot):
+            # Combine the two sub-queries into a query of the form `(first) AND NOT (second)`.
+            subquery_a = self.build_search_query_content(query.subquery_a, config=config)
+            subquery_b = self.build_search_query_content(query.subquery_b, config=config)
+            combined_query = subquery_a._combine(subquery_b, 'NOT')
+            return combined_query
+
+        elif isinstance(query, (And, Or)):
+            subquery_lexemes = [
+                self.build_search_query_content(subquery, config=config)
+                for subquery in query.subqueries
+            ]
+
+            is_and = isinstance(query, And)
+
+            if is_and:
+                return reduce(lambda a, b: a & b, subquery_lexemes)
+            else:
+                return reduce(lambda a, b: a | b, subquery_lexemes)
+
+        raise NotImplementedError(
+            '`%s` is not supported by the SQLite search backend.'
+            % query.__class__.__name__)
+
+    def build_search_query(self, query, config=None):
+        if isinstance(query, MatchAll):
+            return query
+        if isinstance(query, Not):
+            unwrapped_query = query.subquery
+            built_query = Not(self.build_search_query(unwrapped_query, config=config))  # We don't take the Not operator into account.
+        else:
+            built_query = self.build_search_query_content(query, config=config)
+        return built_query
+
+    def build_tsrank(self, vector, query, config=None, boost=1.0):
+        if isinstance(query, (Phrase, PlainText, Not)):
+            rank_expression = BM25()
+
+            if boost != 1.0:
+                rank_expression *= boost
+
+            return rank_expression
+
+        elif isinstance(query, And):
+            return MUL(
+                1 + self.build_tsrank(vector, subquery, config=config, boost=boost)
+                for subquery in query.subqueries
+            ) - 1
+
+        elif isinstance(query, Or):
+            return ADD(
+                self.build_tsrank(vector, subquery, config=config, boost=boost)
+                for subquery in query.subqueries
+            ) / (len(query.subqueries) or 1)
+
+        raise NotImplementedError(
+            '`%s` is not supported by the SQLite search backend.'
+            % query.__class__.__name__)
+
+    def get_index_vectors(self):
+        return [
+            (F('index_entries__title'), F('index_entries__title_norm')),
+            (F('index_entries__body'), 1.0),
+        ]
+
+    def get_search_vectors(self):
+        return self.get_index_vectors()
+
+    def _build_rank_expression(self, vectors, config):
+
+        # TODO: Come up with my own expression class that compiles down to bm25
```

Comment:
TODO

Reply:
Yes, the TODO is there, but because ranking is not yet supported in the SQLite backend. Do you think I should remove that notice? Maybe I could modify the function to just raise a `NotImplementedError`.